auth_checker/models/models.py

```diff
@@ -129,10 +129,41 @@ def render(self):
         }
 
 
-class TokenValidator:
+class BaseTokenValidator:
+    def __init__(self, *args, **kwargs):
+        self.account = None
+        self.token = None
+
+    def _validate(self) -> bool:
+        """
+        Validates a JSON Web Token for this app.
+        :returns tuple: (bool, Account)
+        """
+        if not self.token:
+            raise HTTPException(401, detail="Token is missing")
+        try:
+            if token_map := jot.decode(self.token, JWT_SECRET, algorithms=[JWT_ALGORITHM]):
+                self.account = Account(token_map)
+                return True
+        except jot.exceptions.ExpiredSignatureError:
+            raise HTTPException(401, detail="Token is expired")
+        except jot.exceptions.InvalidSignatureError:
+            raise HTTPException(
+                400, detail="Token has an invalid signature. Check the JWT_SECRET variable."
+            )
+
+
+class RefreshTokenValidator(BaseTokenValidator):
+    def __init__(self, body: AuthnTokenRequestBody):
+        super().__init__()
+        if body:
+            self.token = body.token
```

Comment:
Looks like `body` can't ever be falsey here, we can maybe skip the `if` statement